src/index.ts

```diff
@@ -145,12 +161,28 @@ const useDimensions = <T extends HTMLElement>({
         onResizeRef.current(e);
       }
 
-      setState({
+      const next = {
         currentBreakpoint: e.currentBreakpoint,
         width,
         height,
         entry,
-      });
+      };
+      if (customSetStateRef.current) {
+        setState((s) =>
+          customSetStateRef.current ? customSetStateRef.current(s, next) : next
+        );
+        return;
+      }
+      if (breakpoints && onlyUpdateOnBreakpointChange) {
+        setState((s) => {
```

Comment:
@runar-rkmedia We can simplify the code as below:

```js
if (breakpoints && onlyUpdateOnBreakpointChange ) {
  setState((s) => s.currentBreakpoint === e.currentBreakpoint ? s : next);
  return;
}
```

Reply:
I'll update with your suggestions here.